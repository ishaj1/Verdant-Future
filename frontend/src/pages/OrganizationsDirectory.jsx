import { useEffect } from "react";
import Header from "../components/Header";
import OrganizationCard from "../components/OrganizationCard";
import { useNavigate } from "react-router-dom";

const test_orgs = [
  {
    name: "proj1",
    description: "desc1",
    id: 1,
  },
  {
    name: "comp2",
    description: "desc2",
    id: 2,
  },
];

export default function OrganizationsDirectory({
  isLoggedIn,
  setIsLoggedIn,
  setUserInformation,
  userInformation,
}) {
  const navigate = useNavigate();

  useEffect(() => {
    if (!isLoggedIn) navigate("/");
  });
  return (
    <>
      <Header
        isLoggedIn={isLoggedIn}
        setIsLoggedIn={setIsLoggedIn}
        setUserInformation={setUserInformation}
        userInformation={userInformation}
      />
      {test_orgs.map((org, index) => (
        <OrganizationCard
          description={org.description}
          id={org.id}
          key={index}
          name={org.name}
        />
      ))}
    </>
  );
}
