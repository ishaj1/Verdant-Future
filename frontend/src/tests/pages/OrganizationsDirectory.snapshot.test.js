import React from "react";
import renderer from "react-test-renderer";
import OrganizationsDirectory from "../../pages/OrganizationsDirectory";

// Mock axios to prevent actual HTTP requests during testing
jest.mock("../../api/axios", () => ({
  get: jest.fn(),
}));

// Mock Header component
jest.mock("../../components/Header", () => () => <div>Mock Header</div>);

describe("Organizations Directory Component", () => {
  it("matches the snapshot", () => {
    // Mock data for projects and companies
    const projectsData = [
      { project_name: "Project 1", project_details: "Details 1", project_username: "project1" },
      { project_name: "Project 2", project_details: "Details 2", project_username: "project2" },
    ];

    const companiesData = [
      { company_name: "Company 1", company_details: "Details 1", company_username: "company1" },
      { company_name: "Company 2", company_details: "Details 2", company_username: "company2" },
    ];

    // Render the component tree
    const tree = renderer.create(<OrganizationsDirectory />).toJSON();

    // Update component state with mock data
    tree.props.projectsData = projectsData;
    tree.props.companiesData = companiesData;

    // Expect the rendered component tree to match the snapshot
    expect(tree).toMatchSnapshot();
  });
});
