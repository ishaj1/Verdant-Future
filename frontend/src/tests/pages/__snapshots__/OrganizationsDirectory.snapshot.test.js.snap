// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organizations Directory Component matches the snapshot 1`] = `
<div
  companiesData={
    [
      {
        "company_details": "Details 1",
        "company_name": "Company 1",
        "company_username": "company1",
      },
      {
        "company_details": "Details 2",
        "company_name": "Company 2",
        "company_username": "company2",
      },
    ]
  }
  projectsData={
    [
      {
        "project_details": "Details 1",
        "project_name": "Project 1",
        "project_username": "project1",
      },
      {
        "project_details": "Details 2",
        "project_name": "Project 2",
        "project_username": "project2",
      },
    ]
  }
>
  Mock Header
</div>
`;
